tree/tree/src/TEntryList.cxx

```diff
@@ -287,8 +287,8 @@ TEntryList::TEntryList(const TEntryList &elist) : TNamed(elist)
    fFileName = elist.fFileName;
    fStringHash = elist.fStringHash;
    fTreeNumber = elist.fTreeNumber;
-   fLastIndexQueried = -1;
-   fLastIndexReturned = 0;
+   fLastIndexQueried = elist.fLastIndexQueried;
+   fLastIndexReturned = elist.fLastIndexReturned;
```

Comment:
It is a bit surprising changing for the usual starting point to the previously visited entry fixes the problem.  Those 2 data members are meant to be accelerator and so the default value should have been fine. ...  What is the mechanism leading to the failure?

Reply:
I guess it was an empirical fix. Without this, the returned value is PreviousVisitedEntry + 1, so 16 even if Index is -1
It seems as if the variable index is -1 but the entry pointer is still at 15.